clients/fides-js/src/components/notices/NoticeOverlay.tsx

```diff
@@ -53,37 +71,102 @@ const NoticeOverlay: FunctionComponent<OverlayProps> = ({
     return [];
   };
 
+  /**
+   * Determine which ExperienceConfig translation is being used based on the
+   * current locale and memo-ize it's history ID to use for all API calls
+   */
+  const privacyExperienceConfigHistoryId: string | undefined = useMemo(() => {
+    if (experience.experience_config) {
+      const bestTranslation = selectBestExperienceConfigTranslation(
+        i18n,
+        experience.experience_config
+      );
+      return bestTranslation?.privacy_experience_config_history_id;
+    }
+    return undefined;
+  }, [experience, i18n]);
+
+  /**
+   * Collect the given PrivacyNotices into a list of "items" for rendering.
+   *
+   * Each "item" includes both:
+   * 1) notice: The PrivacyNotice itself with it's properties like keys,
+   *    preferences, etc.
+   * 2) bestTranslation: The "best" translation for the notice based on the
+   *    current locale
+   *
+   * We memoize these together to avoid repeatedly figuring out the "best"
+   * translation on every render, since it will only change if the overall
+   * locale changes!
+   */
+  const privacyNoticeItems: PrivacyNoticeItem[] = useMemo(
+    () =>
+      (experience.privacy_notices || []).map((notice) => {
+        const bestTranslation = selectBestNoticeTranslation(i18n, notice);
+        return { notice, bestTranslation };
+      }),
+    [experience.privacy_notices, i18n]
+  );
+
   const [draftEnabledNoticeKeys, setDraftEnabledNoticeKeys] = useState<
-    Array<PrivacyNotice["notice_key"]>
+    Array<string>
```

Comment:
is there a "safer" way to type this?

Reply:
It's definitely a subjective one... in general, these keys are from a JSON API and therefore they are strings. We could do some clever string-typing to make them a real string (e.g. not empty string) and Typescript is pretty good at this!

I didn't like that we were doing the type "algebra" here and fetching the `PrivacyNotice["notice_key"]` type indirectly, I find that kind of thing really increases the cognitive burden of reading the code. The `PrivacyNotice["notice_key"]` type is a `string` after all, so this just showed that indirectly...

If I was being really clever I might compromise with this:
```typescript
export type PrivacyNoticeKey = PrivacyNotice["notice_key"];
```

That's equivalent (`PrivacyNoticeKey` is a `string`) but is a bit more legible, maybe? I dunno.